x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/components/tables/meta_engines_table_logic.test.ts

```diff
@@ -0,0 +1,246 @@
+
+        expect(MetaEnginesTableLogic.actions.displayRow).toHaveBeenCalled();
+      });
+
+      it('calls fetchSourceEngines when it needs to fetch data for the itemId', () => {
+        mount();
+        jest.spyOn(MetaEnginesTableLogic.actions, 'fetchSourceEngines');
+
+        MetaEnginesTableLogic.actions.fetchOrDisplayRow('test-engine-1');
+
+        expect(MetaEnginesTableLogic.actions.fetchSourceEngines).toHaveBeenCalled();
+      });
+    });
+
+    describe('fetchSourceEngines', () => {
+      it('calls addSourceEngines and displayRow when it has retrieved all pages', async () => {
+        mount();
+        http.get.mockReturnValueOnce(
+          Promise.resolve({
+            meta: {
+              page: {
+                total_pages: 1,
+              },
+            },
+            results: [{ name: 'source-engine-1' }, { name: 'source-engine-2' }],
+          })
+        );
+        jest.spyOn(MetaEnginesTableLogic.actions, 'displayRow');
+        jest.spyOn(MetaEnginesTableLogic.actions, 'addSourceEngines');
+
+        MetaEnginesTableLogic.actions.fetchSourceEngines('test-engine-1');
+        await nextTick();
+
+        expect(http.get).toHaveBeenCalledWith(
+          '/api/app_search/engines/test-engine-1/source_engines',
+          {
+            query: {
+              'page[current]': 1,
+              'page[size]': 25,
+            },
+          }
+        );
+        expect(MetaEnginesTableLogic.actions.addSourceEngines).toHaveBeenCalledWith({
+          'test-engine-1': [{ name: 'source-engine-1' }, { name: 'source-engine-2' }],
+        });
+        expect(MetaEnginesTableLogic.actions.displayRow).toHaveBeenCalledWith('test-engine-1');
+      });
+
+      it('display a flash message on error', async () => {
+        http.get.mockReturnValueOnce(Promise.reject());
+        mount();
+
+        MetaEnginesTableLogic.actions.fetchSourceEngines('test-engine-1');
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalledTimes(1);
+      });
+
+      it('recursively fetches a number of pages', async () => {
+        mount();
+        jest.spyOn(MetaEnginesTableLogic.actions, 'addSourceEngines');
+
+        // First page
+        http.get.mockReturnValueOnce(
+          Promise.resolve({
+            meta: {
+              page: {
+                total_pages: 2,
+              },
+            },
+            results: [{ name: 'source-engine-1' }],
+          })
+        );
+
+        // Second and final page
+        http.get.mockReturnValueOnce(
+          Promise.resolve({
+            meta: {
+              page: {
+                total_pages: 2,
+              },
+            },
+            results: [{ name: 'source-engine-2' }],
+          })
+        );
+
+        MetaEnginesTableLogic.actions.fetchSourceEngines('test-engine-1');
+        await nextTick();
+        await nextTick(); // I think I need this for multiple http.get calls
```

Comment:
Looks like this was missed

```suggestion
        await nextTick();
```